pyiceberg/table/__init__.py

```diff
@@ -3011,9 +3088,11 @@ def __init__(
         io: FileIO,
         commit_uuid: Optional[uuid.UUID] = None,
         snapshot_properties: Dict[str, str] = EMPTY_DICT,
+        only_delete_within_latest_spec: bool = False,
```

Comment:
Can you give more context around this, and why it is needed?
```suggestion
        only_delete_current_spec: bool = False,
```
The latest can be different from the current spec. If you add a field to the partition spec and drop it immediately, it will re-use the existing spec.